pkg/controller/vsphere/reconciler.go

```diff
@@ -507,3 +512,43 @@ func (vm *virtualMachine) getNetworkStatusList(client *vim25.Client) ([]NetworkS
 
 	return networkStatusList, nil
 }
+
+// IgnitionConfig returns a slice of option values that set the given data as
+// the guest's ignition config.
+func IgnitionConfig(data []byte) []types.BaseOptionValue {
+	config := EncodeIgnitionConfig(data)
+
+	if config == "" {
+		return nil
+	}
+
+	return []types.BaseOptionValue{
+		&types.OptionValue{
+			Key:   "guestinfo.ignition.config.data",
+			Value: config,
+		},
+		&types.OptionValue{
+			Key:   "guestinfo.ignition.config.encoding",
+			Value: "base64",
+		},
+	}
+}
+
+// EncodeIgnitionConfig attempts to decode the given data until it looks to be
+// plain-text, then returns a base64 encoded version of that plain-text.
```

Comment:
nit: this is not ignition specific, could be may be called `base64DataToString`, `sanitiseBase64` or similar.

Reply:
The implementation happens to not be Ignition specific, but the function is meant to encode Ignition data for inclusion in guestinfo variables, so I think the name should reflect that.